Adding Phobos station

Starbuckss14 opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
This is DarkDan's map Phobos! A mid sized map featuring a asteroid theme!


- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase




**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

<!--
Make sure to take this Changelog template out of the comment block in order for it to show up.
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
add: added Phobos station!

Mangohydra commented:
best map ever

lzk228 commented:
bagel 3.0??

UbaserB commented:
feels like a lot of weirdly empty space

Starbuckss14 commented:
> best map ever

Why thank you!

Starbuckss14 commented:
> bagel 3.0??

Lol It kind of looks like that

Starbuckss14 commented:
> feels like a lot of weirdly empty space

sorry what part?

IProduceWidgets commented:
Are the candles the decorative / always lit variety? It seems like a lot of candles. I'm not against candles, I just noticed the quantity.

Perma having an actual non-joke disposal even though it goes to space seems kinda uhhhhh.

No meteor protection (lattice / grilles) on an already very spaceable map.

Free gun in the bar!?

Clown HOS feels very NRP.

I don't like wall mounts on rock. There's a lot on the north side including lockers/closets and apcs.

I wanna say probably don't map the bananium horn. 

Would like others thoughts on the sauna. I fear it may just be a death box.

Bridge hallway emergency lockers feel shoehorned and probably just need a small recess to sit in.

Why are there syndicate airlocks on the station (in the north)???

I couldn't find where your disposals room is but maybe I'm blind or weh. 

Disclaimer: I did not load the map ingame, and so did not check wiring/power/pipes/atmos/dispos or anything else that would require that.

Hmeister-real commented:
my thoughts: map looks fun, not many flaws i can point out, but a few are that
1) That Bar revolver should be in a locked Barkeep accessable windoor 1x1 chamber behind the counter, or jusr in the backroom, not open for a tider to rush to round start
2) the map looks extremely spacable, so engi should spawn with a ton more at, the very least, inflatable walls if not more.
3) the moment a syndie/shitter get a pickaxe and an EVA its over.

Errant-4 commented:
Can you please the PR template again. Thanks

Errant-4 commented on Resources/Prototypes/Maps/Pools/default.yml:
```diff
@@ -14,3 +14,4 @@
   - Saltern
   - Packed
   - Reach
+  - phobos
```
Literally every other map in this list is capitalised, and in alphabetical order

Starbuckss14 replied:
I will fix that

Errant-4 replied:
For the record, you are supposed to Resolve reviews _after_ you have deployed the fix

Errant-4 commented on Resources/Prototypes/Maps/phobos.yml:
```diff
@@ -0,0 +1,60 @@
+- type: gameMap
+  id: phobos
+  mapName: 'phobos'
```
'Phobos'

Errant-4 replied:
Okay my mistake for not being more specific. As far as I can see, only this specific string, `mapName`, should be capitalised. Not id, and not the file's name. Look at the other map files. While blindly copying things has it's own problems, when you are making something new then looking at the existing examples is probably still your best starting point, and consistency is good

Errant-4 commented on Resources/Prototypes/Maps/phobos.yml:
```diff
@@ -0,0 +1,60 @@
+  mapName: 'phobos'
+  mapPath: /Maps/phobos.yml
+  minPlayers: 0
+  maxPlayers: 50
```
I'm no expert but min 0 max 50 seems like an... odd range? Was it intended like this?

Starbuckss14 replied:
I just kinda picked a number. Any suggestions on making it higher/lower?

Errant-4 replied:
I'm not really a mapper, but a map that is targeted for up to 50 players should have a minimum around 15-20 , judging by other map prototypes. Might want to work that out with experienced mappers in Cartography

Starbuckss14 replied:
Map size updated to 65

Errant-4 replied:
I repeat: the issue is that you have configured the map to be an equally valid choice (as far as the game engine is concerned) for 65 players or 4. Do you see the issue with putting 4 players on a map large enough for 65

Starbuckss14 replied:
OK I understand the issue. I will count up the spawns and leave like 10 spots for wiggle room for greys?

Errant-4 replied:
Look at a similar sized map to get an idea for individual role slot numbers, but specifically I meant that you need to set a `minPlayers` value so the map does not get picked if there are too few players in the lobby

CMDR-Piboy314 commented:
![image](https://github.com/space-wizards/space-station-14/assets/92357316/cd53e122-d2a9-45de-af37-bff96da36878)
what the hell is this laser also the fireplace in the wall in the library is pretty horrific

add some lattices around in space and solars out there too, lattices connecting e.g. the bridge thats kind of hanging off there would be nice as well.

adding some medical supply cabinets would be much appreciated and there probably shouldn't be a jug vendor in that hallway to cmo. also the morgue having syndicate airlocks is not a great idea but the syndicate posters and stuff are cool. the sand looks really out of place for the cmo walls.

rd literally cant get out of their room????
![image](https://github.com/space-wizards/space-station-14/assets/92357316/9a7d4e1e-c251-4864-9332-c3421c61e051)

the chef's bedroom thing is really weird and fireplaces being anywhere but under walls looks pretty horrible imo, also why the hell is there an atmos crab there... in some maints there are way too many candles like south.

this is also pretty horrible, there's a tree covering the maints entrance, so many chairs in a row etc.
![image](https://github.com/space-wizards/space-station-14/assets/92357316/2e13726f-9b48-49b3-9736-5587cddde164)

mirrorcult commented:
> rd literally cant get out of their room???? ![image](https://private-user-images.githubusercontent.com/92357316/304902315-9a7d4e1e-c251-4864-9332-c3421c61e051.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NTM0NzQsIm5iZiI6MTcwNzk1MzE3NCwicGF0aCI6Ii85MjM1NzMxNi8zMDQ5MDIzMTUtOWE3ZDRlMWUtYzI1MS00ODY0LTkzMzItYzM0MjFjNjFlMDUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDIzMjYxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxMjg4M2RiOWU5NzE4MGY2YjM4MmIwODQyZGQ1MTUyMGI2NTc5ZmEzMDM2Y2QzYzhkMjNmYzc4YTg2YWM1ZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.inwORSe25OfbyLVUzZAPIEwIxrRCFs9QGUSJ8X-hrU0)
> 

pretty sure this is just a maprenderer bug, it should be 1 tile up in game

UbaserB commented:
> > rd literally cant get out of their room???? ![image](https://private-user-images.githubusercontent.com/92357316/304902315-9a7d4e1e-c251-4864-9332-c3421c61e051.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NTM0NzQsIm5iZiI6MTcwNzk1MzE3NCwicGF0aCI6Ii85MjM1NzMxNi8zMDQ5MDIzMTUtOWE3ZDRlMWUtYzI1MS00ODY0LTkzMzItYzM0MjFjNjFlMDUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDIzMjYxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxMjg4M2RiOWU5NzE4MGY2YjM4MmIwODQyZGQ1MTUyMGI2NTc5ZmEzMDM2Y2QzYzhkMjNmYzc4YTg2YWM1ZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.inwORSe25OfbyLVUzZAPIEwIxrRCFs9QGUSJ8X-hrU0)
> 
> pretty sure this is just a maprenderer bug, it should be 1 tile up in game

same with the trees

Starbuckss14 commented:
> > rd literally cant get out of their room???? ![image](https://private-user-images.githubusercontent.com/92357316/304902315-9a7d4e1e-c251-4864-9332-c3421c61e051.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NTM0NzQsIm5iZiI6MTcwNzk1MzE3NCwicGF0aCI6Ii85MjM1NzMxNi8zMDQ5MDIzMTUtOWE3ZDRlMWUtYzI1MS00ODY0LTkzMzItYzM0MjFjNjFlMDUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE0VDIzMjYxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxMjg4M2RiOWU5NzE4MGY2YjM4MmIwODQyZGQ1MTUyMGI2NTc5ZmEzMDM2Y2QzYzhkMjNmYzc4YTg2YWM1ZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.inwORSe25OfbyLVUzZAPIEwIxrRCFs9QGUSJ8X-hrU0)
> 
> pretty sure this is just a maprenderer bug, it should be 1 tile up in game

Ya its a bug in the render

Starbuckss14 commented:
> ![image](https://private-user-images.githubusercontent.com/92357316/304901240-cd53e122-d2a9-45de-af37-bff96da36878.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NTY1MDcsIm5iZiI6MTcwNzk1NjIwNywicGF0aCI6Ii85MjM1NzMxNi8zMDQ5MDEyNDAtY2Q1M2UxMjItZDJhOS00NWRlLWFmMzctYmZmOTZkYTM2ODc4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE1VDAwMTY0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkwM2QyYzNjOTZhM2NmNzExN2YxMjNhNGY1NjU5MmMwMDY4NWU2NDM0NWFlMWI4Y2MwOWRkMGZiMGQ5MThkN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jzNvp_OOzGyvo23mT3Q6fYHXwFy7pyYhy6qh8UpcZd0) what the hell is this laser also the fireplace in the wall in the library is pretty horrific
> 
> add some lattices around in space and solars out there too, lattices connecting e.g. the bridge thats kind of hanging off there would be nice as well.
> 
> adding some medical supply cabinets would be much appreciated and there probably shouldn't be a jug vendor in that hallway to cmo. also the morgue having syndicate airlocks is not a great idea but the syndicate posters and stuff are cool. the sand looks really out of place for the cmo walls.
> 
> rd literally cant get out of their room???? ![image](https://private-user-images.githubusercontent.com/92357316/304902315-9a7d4e1e-c251-4864-9332-c3421c61e051.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NTY1MDcsIm5iZiI6MTcwNzk1NjIwNywicGF0aCI6Ii85MjM1NzMxNi8zMDQ5MDIzMTUtOWE3ZDRlMWUtYzI1MS00ODY0LTkzMzItYzM0MjFjNjFlMDUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE1VDAwMTY0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk4MzNlZjJiZmI3Njk5NDAxM2NhZDMyMDI5NDY2MDNlYzVhNjY3YWNhZmUzMzM1ODAwZWFkYjdhODAwYjEyNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2wDKkT-u_58R8k_J3hc7ZR6Fhzf0b06iz1yYMn54aTk)
> 
> the chef's bedroom thing is really weird and fireplaces being anywhere but under walls looks pretty horrible imo, also why the hell is there an atmos crab there... in some maints there are way too many candles like south.
> 
> this is also pretty horrible, there's a tree covering the maints entrance, so many chairs in a row etc. ![image](https://private-user-images.githubusercontent.com/92357316/304902786-2e13726f-9b48-49b3-9736-5587cddde164.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NTY1MDcsIm5iZiI6MTcwNzk1NjIwNywicGF0aCI6Ii85MjM1NzMxNi8zMDQ5MDI3ODYtMmUxMzcyNmYtOWI0OC00OWIzLTk3MzYtNTU4N2NkZGRlMTY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE1VDAwMTY0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlOGI4YTNkZWMxNDliYjczMThiMzk0MWYzOWY5NGQzNzIzYmE3Yjg3YjAyOTQ2ZWMzNmIzYWNmZDQ3NTEwZjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tSFdNEbimJW2WX9AE74xeuhQ1uRgO7Fau_TU2nJ9vZo)

Laser removed. Med tweaked. Hydro given a pipe system to the chef and chem. Map render bug. Fireplace fixed.

Starbuckss14 commented:
> Can you please the PR template again. Thanks
> 
> (If it gets merged then the description you provide will also be used on the wiki so it would be helpful to have something)
> 
> Putting a map in rotation also needs to be mentioned in the Changelog

Fixed

Starbuckss14 commented:
Is there a way for me to make a new render of my map so the render is up to date?

Errant-4 commented:
> Is there a way for me to make a new render of my map so the render is up to date?

`dotnet run --project Content.MapRenderer Phobos`

Starbuckss14 commented:
> > Is there a way for me to make a new render of my map so the render is up to date?
> 
> `dotnet run --project Content.MapRenderer Phobos`

Where do I run that command?

Errant-4 commented:
> > > Is there a way for me to make a new render of my map so the render is up to date?
> > 
> > 
> > `dotnet run --project Content.MapRenderer Phobos`
> 
> Where do I run that command?

In any command line interface (that is currently focused on the directory of your project). Your IDE should have a terminal built in, and will already be focused on your project files. Or you can use the one provided by your OS

Starbuckss14 commented:
> > > > Is there a way for me to make a new render of my map so the render is up to date?
> > > 
> > > 
> > > `dotnet run --project Content.MapRenderer Phobos`
> > 
> > 
> > Where do I run that command?
> 
> In any command line interface (that is currently focused on the directory of your project). Your IDE should have a terminal built in, and will already be focused on your project files. Or you can use the one provided by your OS

I am having issues with getting the render to work. Is there any documentation on it?

Starbuckss14 commented:
> dotnet run --project Content.MapRenderer Phobos

I found my issue the map renderer is out of date and wont work because of the newer prototypes. I will take some photos instead

Starbuckss14 commented:
I made some changes based on the suggestions. Below are some images of the map. Let me know what else needs fixing before we are ready to merge.

The maints needed some more hiding spots so I added this new "old bridge" 
![Screenshot (43)](https://github.com/space-wizards/space-station-14/assets/145926356/bb25317e-22cb-4a5b-85ea-ad65ea97059c)
Changed the HOS's office and removed the clown mask
![Screenshot (44)](https://github.com/space-wizards/space-station-14/assets/145926356/36e44823-8d2c-48f1-9d2b-fb883d5deee4)
Added a pipe system for hydro to send and receive items from the chef and chem 
![Screenshot (45)](https://github.com/space-wizards/space-station-14/assets/145926356/2cd00a33-f9d7-49e7-ac8e-11e10abb8b32)
![Screenshot (46)](https://github.com/space-wizards/space-station-14/assets/145926356/f362513e-bd34-4766-bb27-
751152b84ad1)
updated the abandoned syndi shuttle to have medical airlocks
![Screenshot (47)](https://github.com/space-wizards/space-station-14/assets/145926356/67c99ba8-71a4-4d9b-8b70-dd1a23254240)

MjrLandWhale commented:
Here's my feedback for what it's worth:

1. There are far too many space facing walls on the west side which are not reinforced. All other stations (to my knowledge) exclusively use reinforced walls when separating station from space. In my mind this should also apply where the only additional support is something like rocks. As an example near the science department, anything marked in red _should_ be reinforced in my mind
![image](https://github.com/space-wizards/space-station-14/assets/12506666/4ec33776-7c21-4847-a69e-0d1886f6948b)

2. The inclusion of many vending machines directly in the hallways feels somewhat... clunky to me. My recommendation is to recess these sections one tile where possible. Something like we see in other stations (origin here as an example):
![image](https://github.com/space-wizards/space-station-14/assets/12506666/4fc69b48-312e-4695-8ed3-e2131aafe6e5)
You already take this approach on the eastern main hallway, where you elect to go with a 4 wide, using the extra space for chairs, tables, etc. 

3. Further compounding the issues from #1 is the lack of any maintenance-based firelocks to control loss of atmosphere.

I think overall I have major concerns about the survivability of this station from sabotage. With only a welder, wrench, and pickaxe I can:

- break 8 rocks and space most of medical. Also deconstruct 2 normal walls and you've spaced the main hall outside med and the hallway leading to science.
- Deconstruct 2 normal walls to space the main hallway (and maints) by sec & cargo, effectively cutting off security unless they have a suit or run around the entire station.
- Break 2 rocks and space dorms.
- Deconstruct 2 normal walls and plasma flood the entire hallway outside engineering while also venting the station's entire plasma gas supply. (side note: why in the world is the plasma gas hooked directly into distro via a manual valve?)

Unless we have a way of giving nearly everyone hardsuits right off the bat, I fear this station in its current form will result in a rapid, widespread loss of atmosphere. This will have the consequence of killing/injuring far too much of the station without any meaningful way to control or stop it.

Emisse commented:
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.
